src/app/components/candidate-card-popup/candidate-card-popup/candidate-card-popup.component.ts

```diff
@@ -1,110 +1,267 @@
-  ];
+  public checked = false;
 
   cancel(): void {
     this.modal.emit();
   }
 
+  addStatus(): void {
+    if (this.checked) {
+      this.candidateStatus = this.selectedValue;
+      this.counter++;
+      console.log(this.selectedValue);
+      console.log('new status', this.candidateStatus);
+
+      this.checked = false;
+      console.log(this.checked);
+    } else {
+      this.counter--;
+      this.checked = true;
+      console.log(this.checked);
+    }
+  }
+
+  changeStatus(): void {
+    const STATUS = {
+      id: this.CANDIDATES_INFO.id,
+      candidateSandboxId: this.candidateSandbox,
+      newStatusId: '',
+    };
+    console.log(this.statuses);
+    this.statuses.filter((el: any) => {
+      el.name === this.candidateStatus ? (STATUS.newStatusId = el.id) : null;
+    });
+    console.log(STATUS);
+
+    this.putStatus(STATUS);
+  }
+
+  putStatus(STATUS: { id: string; candidateSandboxId: string; newStatusId: string }): any {
+    return axios
+      .put(
+        `${this.URL}candidates/${STATUS.id}/candidatesandboxes/${STATUS.candidateSandboxId}?newStatusId=${STATUS.newStatusId}`,
+      )
+      .then((response: any) => console.log(response))
+      .catch((error: any) => console.log(error));
+  }
+
+  printForm(): any {
+    const FEEDBACK_PUT = {
+      grade: this.sliderValue,
+      userReview: this.userReview,
+    };
+    const FEEDBACK_POST = {
+      userId: this.userId,
+      grade: this.sliderValue,
+      userReview: this.userReview,
+      candidateProccesId: this.candidateProccesId,
+    };
+    !this.feedbackId ? console.log(FEEDBACK_POST) : console.log(FEEDBACK_PUT);
+    !this.feedbackId ? this.postFeedbacks(FEEDBACK_POST) : this.putFeedbacks(FEEDBACK_PUT);
+  }
   onChangeRange(rangeValue: MatSliderChange): void {
     this.sliderValue = rangeValue.value ? rangeValue.value : 0;
   }
 
-  getCandidateInfo(): any {
+  postFeedbacks(FEEDBACK_POST: {
+    userId: any;
+    grade: number;
+    userReview: string;
+    candidateProccesId: string;
+  }): any {
+    return axios
+      .post(`${this.URL}feedbacks`, FEEDBACK_POST)
+      .then((response: any) => console.log(response))
+      .catch((error: any) => console.log(error));
+  }
+
+  getFeedbacks(): any {
     return axios
-      .get(`https://jsonplaceholder.typicode.com/users`)
-      .then((response: any) => console.log('getC', response))
-      .catch((error: any) => console.log('getC', error));
+      .get(`${this.URL}feedbacks/${this.feedbackId}`)
+      .then((response: any) => console.log(response))
+      .catch((error: any) => console.log(error));
   }
 
-  getUserInfo(): any {
+  getStatuses(): any {
     return axios
-      .get(`https://jsonplaceholder.typicode.com/users`)
-      .then((response: any) => console.log('getU', response.data))
-      .catch((error: any) => console.log('getU', error));
+      .get(`${this.URL}statuses`)
+      .then((response: AxiosResponse<any, any>): void => {
+        setTimeout(() => {
```

Comment:
better done without setTimeout

Reply:
Mb will better to use our auth service or angular httpclient.

Reply:
we need async for hand over to other component